client/java/src/main/java/io/openlineage/utils/DatasetIdentifierUtils.java

```diff
@@ -0,0 +1,92 @@
+/*
+/* Copyright 2018-2023 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.utils;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+public class DatasetIdentifierUtils {
+
+  private static final String DEFAULT_SCHEME = "file";
+  /** The directory separator, a slash, as a character. */
+  public static final char SEPARATOR_CHAR = '/';
+  /** The directory separator, a slash. */
+  public static final String SEPARATOR = "/";
+
+  /** Whether the current host is a Windows machine. */
+  public static final boolean WINDOWS = System.getProperty("os.name").startsWith("Windows");
+
+  /** Pre-org.apache.hadoop.shaded.com.iled regular expressions to detect path formats. */
+  private static final Pattern HAS_DRIVE_LETTER_SPECIFIER = Pattern.compile("^/?[a-zA-Z]:");
+
+  public static String nameFromURI(URI uri) {
```

Comment:
If you had DatasetIdentifier there you'd not have to do `namespaceFromURI` and `nameFromURI` but get both at the same time

Reply:
I know that. Intention was to limit amount of code that need to be changed.